site/src/sphinx/client-retry.rst

```diff
@@ -335,6 +343,56 @@ This will produce only single request and response log pair regardless how many
 
     Please refer to :ref:`nested-log`, if you are curious about how this works internally.
 
+``RetryingClient`` with circuit breaker
+---------------------------------------
+
+You might want to use :ref:`client-circuit-breaker` with :api:`RetryingHttpClient` using decorator_:
+
+.. code-block:: java
+
+    import com.linecorp.armeria.client.circuitbreaker.CircuitBreakerStrategy;
+    import com.linecorp.armeria.client.circuitbreaker.CircuitBreakerHttpClientBuilder;
+
+    CircuitBreakerStrategy cbStrategy = CircuitBreakerStrategy.onServerErrorStatus();
+    RetryStrategy myRetryStrategy = new RetryStrategy(){ ... };
+
+    HttpClient client = new HttpClientBuilder(...)
+            .decorator(new CircuitBreakerHttpClientBuilder(cbStrategy).newDecorator())
+            .decorator(new RetryingHttpClientBuilder(myRetryStrategy).newDecorator())
+            .build();
+
+    final AggregatedHttpMessage res = client.execute(...).aggregate().join();
+
+This decorates :api:`CircuitBreakerHttpClient` with :api:`RetryingHttpClient` so that the :api:`CircuitBreaker`
+judges every request and retried request as successful or failed. If the failure rate exceeds a certain
+threshold, it raises a :api:`FailFastException`. When using both clients, you need to write a custom
+:api:`RetryStrategy` to handle this exception so that the :api:`RetryingHttpClient` does not attempt
+a retry unnecessarily when the circuit is open, e.g.
+
+.. code-block:: java
+
+    import com.linecorp.armeria.client.circuitbreaker.FailFastException;
+
+    new RetryStrategy() {
```

Comment:
nit: How about changing this to a named class?

Reply:
I will assign this to `myRetryStrategy` which is used in the above example. Is it OK?